doc/source/dev_guide/custom_reader.rst

```diff
@@ -535,7 +535,15 @@ the :func:`xarray.open_dataset` function in a custom file handler is a much
 better idea.
 
 .. note::
-   Be careful about the data types of the datasets your reader is returning.
+   Be careful about the data types of the dataset attributes your reader is
+   returning.  JSON is used internally in Satpy to serialize the function
+   arguments so we can hash them and then use that hash as the filename for the
+   cached results. Numpy types don't serialize into JSON, therefore please make
+   sure the attributes being set upon reading does not contain numpy scalar
+   types.
+
+.. note::
+   Also, be careful about the data types of the datasets your reader is returning.
```

Comment:
This is a note. In my opinion there is no "before" or "after" to other content so "Also" seems odd here.

```suggestion
   Be careful about the types of the data your reader is returning.
```